app/components/hearings-list-for-milestones-list.js

```diff
@@ -0,0 +1,66 @@
+import { computed } from '@ember/object';
+
+export default class ProjectMilestoneComponent extends Component {
+  // @argument
+  milestone;
+
+  milestoneParticipantReviewLookup = {
+    BP: 'Borough President Review',
+    BB: 'Borough Board Review',
+    CB: 'Community Board Review',
+  }
+
+  // An array of disposition models that match the current milestone
+  @computed('milestone', 'milestone.project.dispositions')
+  get currentMilestoneDispositions() {
+    const currentMilestoneDispositions = [];
+
+    const milestone = this.get('milestone');
+    // ALL dispositions associated with a milestone's project
+    const dispositions = milestone.get('project.dispositions');
+    const milestoneParticipantReviewLookup = this.get('milestoneParticipantReviewLookup');
+
+    // iterate through ALL of the current project's dispositions
+    // IF a single disposition's assignment's dcpLupteammemberrole matches the
+    // current milestone's displayName, push that disposition into the new
+    // currentMilestoneDispositions array
+    dispositions.forEach(function(disposition) {
+      const participantType = disposition.get('assignment.dcpLupteammemberrole');
+
+      // milestoneParticipantReviewLookup[participantType] = e.g. `Community Board Review`
+      if (milestone.displayName === milestoneParticipantReviewLookup[participantType]) {
+        currentMilestoneDispositions.push(disposition);
+      }
+    });
+
+    // if dispositions have been pushed to the array (its length is greater than 1)
+    // return arrray -- otherwise return null
+    if (currentMilestoneDispositions.length < 1) {
+      return null;
+    } return currentMilestoneDispositions;
+  }
+
+  // An array of disposition models that match the current milestone
+  @computed('milestone.milestonename')
+  get currentMilestoneUserType() {
+    const milestoneName = this.get('milestone.milestonename');
+    return milestoneName.replace(' Review', '');
+  }
+
+
+  // if each dcpPublichearinglocation and dcpDateofpublichearing properties are filled in currentMilestoneDispositions array,
+  // then hearings have been submitted for that project
+  @computed('currentMilestoneDispositions.@each.{dcpPublichearinglocation,dcpDateofpublichearing}')
+  get hearingsSubmitted() {
+    const dispositions = this.get('currentMilestoneDispositions');
+    // array of hearing locations
+    const dispositionHearingLocations = dispositions.map(disp => `${disp.dcpPublichearinglocation}`);
+    // array of hearing dates
+    const dispositionHearingDates = dispositions.map(disp => disp.dcpDateofpublichearing);
+    // hearingsSubmittedForProject checks whether each item in array is truthy
+    const arraysFilled = dispositionHearingLocations.length > 0 && dispositionHearingDates.length > 0;
+    const hearingsSubmittedForProject = arraysFilled && dispositionHearingLocations.every(location => !!location) && dispositionHearingDates.every(date => !!date);
+    return hearingsSubmittedForProject;
```

Comment:
~hmm what's the requirement? you want to make sure that _all_ hearing locations and dates are filled out? it looks like this would only check if at least _some_ of them are, not _all_.~

ah okay, so you're using `every` to check — great! I think you could simplify this _a lot_ by always returning an array from the `currentMilestoneDispositions` computed prop. When I'm in these situations, I always try to do everything I can to make sure that the computed I'm writing always consistently returns the same primitive. This isn't _always_ possible, but it does help with debugging. Because it looks like your `arraysFilled` here is checking that they're arrays? Not quite sure... At any rate, you could probably just do this:

```javascript
return this.currentMilestoneDispositions.every((disposition) => {
  const { dcpPublichearinglocation, dcpDateofpublichearing } = disposition;

  // no need to coerce to boolean with double bang—these are ember model attrs
  //and are expected to be strings/null, which are implicitly coercible to boolean.
  return dcpPublichearinglocation && dcpPublichearinglocation;
});
```

you could even get extra fancy with destructuring out of the function signature:
```javascript
return this.currentMilestoneDispositions
  .every(({ dcpPublichearinglocation, dcpPublichearinglocation }) => dcpPublichearinglocation && dcpPublichearinglocation);
```

Reply:
Okay cool, will explore :)